### ISSUE TITLE
Template biding of ViewChild property is one change detection behind

### ISSUE THREAD

**Issue by Totati** [angular/angular#51721]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

When I try to bind a ViewChild property into the template (as string interpolation or to an input of another component (not in reproduction as it does not matter)) it is one change detection cycle behind.  Using `markForCheck` in a `setTimeout` solves the problem.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-qxtyxr

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular 16.2.4
```

**Comment by JeanMeche** [comment]
That the pretty much expected : you need 2 CD cycles to reach the right rendering : 

1. You click, the boolean changes the view gets updated (the viewchild gets references/de-referenced
2. This time it will pick the new viewChild value and display it. 

To explain why `setTimeout` works here (and not `Promise.resolve().then` or `queueMicroTaks()`), it's that setTimeout will actually "schedule" a new CD where.

**Comment by Totati** [comment]
Yes, understand that. The usage notes shows the example with a setter that sets the value in a setTimeout, but without explanation. It's unusual that I could not find anything explaining this behaviour.

**Comment by JeanMeche** [comment]
Ok, you're talking about this doc : https://angular.io/api/core/ViewChild 

It definitely needs some explanation why a setTimeout is used here !

**Comment by pkozlowski-opensource** [comment]
Note that this works as-is (without any timeouts or ChangeDetectorRef usage) when using signals: https://stackblitz.com/edit/stackblitz-starters-vhdxmo?file=src%2Fmain.ts 

This is because all state changes _will_ trigger change detection as needed.

Having said this the coding pattern exposed here is somewhat problematic since it uses rendered UI... to drive more UI rendering. This is something that should be avoided and I would suggest structuring this part of the UI differently (hard to be more specific without fully understanding the real-life use-case).

**Comment by JeanMeche** [comment]
Per above, we have both now a better approach with the signal reactivity + there is some design issue in the example.
